### PR TITLE
Add ImageMediaInput

### PR DESCRIPTION
**Pull request by gettinToasty**
Just a filetype filtered version of the media gallery, @holiber i think this will unblock the rest of custom fields for you :)

### CODE REVIEW

File: app/components/shared/inputs/index.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ export interface IListOption<TValue> {
 export interface IMediaGalleryMetadata extends IInputMetadata {
   fileName: string;
```
Comment (holiber) [inline app/components/shared/inputs/index.ts]: Why we need `fileName` here? We already have `value` property

File: app/services/media-gallery/media-gallery.ts [hunk 2]
```diff
@@ -28,7 +28,8 @@ export interface IMediaGalleryInfo extends IMediaGalleryLimits {
 }
 
 interface IMediaGalleryProps {
-  audioOnly: boolean;
+  audioOnly?: boolean;
+  imageOnly?: boolean;
```
Comment (holiber) [inline app/services/media-gallery/media-gallery.ts]: You can't set both, right? Maybe use something like `filter: {images: true, audio: true }` ?

File: app/components/shared/inputs/index.ts (hunk 1 above)
Comment (gettinToasty) [reply app/components/shared/inputs/index.ts]: `fileName` is a display value whereas `value` is the href, but i suppose we could move the filename parsing into the input component that probably makes more sense

File: app/services/media-gallery/media-gallery.ts (hunk 2 above)
Comment (gettinToasty) [reply app/services/media-gallery/media-gallery.ts]: yea that's true, i'll look into making a better pattern here

File: app/components/shared/inputs/index.ts (hunk 1 above)
Comment (holiber) [reply app/components/shared/inputs/index.ts]: Yeah, we probably just need a getter here.

Comment (holiber) [reply app/components/shared/inputs/index.ts]: To be clear

Comment (holiber) [reply app/components/shared/inputs/index.ts]: To be clear, the `metadata` property intended to store only values that are not going to be changed when you change the `value` property.

**Pull request merged**